app/src/main/java/com/duckduckgo/app/browser/IsSwipingTabsFeatureEnabled.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import com.duckduckgo.di.scopes.AppScope
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@SingleInstanceIn(AppScope::class)
+class IsSwipingTabsFeatureEnabled @Inject constructor(
+    swipingTabsFeature: SwipingTabsFeature,
+) {
+    private val isEnabled: Boolean = swipingTabsFeature.self().isEnabled()
```

Comment:
are you sure this doens't call IO right when the class is instantiated? If it does (I think it does) we should change it

Reply:
+1

Reply:
> are you sure this doens't call IO right when the class is instantiated? If it does (I think it does) we should change it.

You mean when the `SwipingTabsFeature` is being initialized for the first time? I don’t think so, at least from what I saw when stepping through the code. But I couldn’t peek inside the store so I may be wrong. We can initialize the variable lazily, though, to be safe.

> Also, I would question why do you need this class anyway, you can more easily do the same (without the overhead of a whole new class) by using lazy in the caller (RealTabManager)

I’m checking for the flag in multiple places (not here but in a subsequent PR), so I need to store the value somewhere for consistency.